### PR TITLE
[mlir][vector] Add more tests for ConvertVectorToLLVM (5/n)

### PR DESCRIPTION
**Pull request by banach-space**
Adds tests with scalable vectors for the Vector-To-LLVM conversion pass.
Covers the following Ops:
  * vector.extract_strided_slice

In practice, this has meant adding 1 new test:
*  `@extract_strided_slice_f32_1d_from_2d_scalable`,

and replacing `@extract_strided_slice_scalable` with 
`@extract_strided_slice_f32_2d_from_2d_scalable`.

For consistency with other tests, I have also removed "vector" from test
functions names for:
  * vector.print
  * vector.type_cast
  * vector.extract_strided_slice

The first two Ops precede `vector.extract_strided_slice` in the test
file, i.e. those should be next to be updated in this series of patches.
However,
  * For `vector.print`, we don't use vectors in this test file (that
    would require running VectorToSCF).
  * For `vector.type_cast`, the existing tests assume fixed-width sizes.
    We need to write new tests and I am leaving that as a TODO.

Note, I've also updated test function names to be more descriptive and
consistent with other tests, e.g.
  * `@extract_strided_slice3` -> `@extract_strided_slice_f32_2d_from_2d`

### CODE REVIEW

**Review by joker-eph — APPROVED** [review]

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir [hunk 1]
```diff
@@ -1699,46 +1705,50 @@ func.func @vector_print_scalar_f64(%arg0: f64) {
...
 //  CHECK-SAME:    %[[A:.*]]: vector<4xf32>)
 //       CHECK:    %[[T0:.*]] = llvm.shufflevector %[[A]], %[[A]] [2, 3] : vector<4xf32>
 //       CHECK:    return %[[T0]] : vector<2xf32>
 
+// NOTE: For scalable vectors we could only extract vector<[4]xf32> from vector<[4]xf32>, but that would be a NOP.
+
 // -----
 
-func.func @extract_strided_index_slice1(%arg0: vector<4xindex>) -> vector<2xindex> {
+func.func @extract_strided_index_slice_index(%arg0: vector<4xindex>) -> vector<2xindex> {
```
Comment (nujaa) [inline mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:1732]:
```suggestion
func.func @extract_strided_slice_index(%arg0: vector<4xindex>) -> vector<2xindex> {
```

**Review by nujaa — COMMENTED** [review]
Welcome back, just a little NIT.

File: mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir (hunk 1 above)
Comment (banach-space) [reply mlir/test/Conversion/VectorToLLVM/vector-to-llvm.mlir:1732]: Your eagle-eye saves me once again - thank you 🙏🏻 :)

**Comment by banach-space** [comment]
> Welcome back, just a little NIT.

Great to be back, though only for this week. I'm OOO next two weeks, but I will monitor my GitHub inbox (should you send sth ;-) )

**Comment by banach-space** [comment]
Ping @nujaa :) Please feel free to land if this looks good to you, thanks!

**Comment by nujaa** [comment]
LGTM!

**Pull request merged**